### ISSUE TITLE
GPS Coordinates inaccurate on https://www.gps-coordinates.net/my-location

### ISSUE THREAD

**Issue by ryanbr** [brave/brave-browser#16897]
## Description 
Open `https://www.gps-coordinates.net/my-location`  in Brave, shows incorrect location compared to other browsers

## Steps to Reproduce

   1. Compare Chrome, Firefox and Brave on `https://www.gps-coordinates.net/my-location`
   2. Brave showed a different location compared to other browsers
   3. 

## Actual result:

Unable to post, due to privacy. But it's off about GPS Coords is off about 14km in my case

## Expected result:
Show the same location results

## Reproduces how often: 

Easily 

## Brave version (brave://version info)

`Version 1.26.74 Chromium: 91.0.4472.124 (Official Build) (64-bit)`

## Version/Channel Information:

- Can you reproduce this issue with the current release? 
- Can you reproduce this issue with the beta channel? 
- Can you reproduce this issue with the nightly channel? 

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? No
- Is the issue reproducible on the latest version of Chrome? No

## Miscellaneous Information:
Tested in Brave release, Beta and Nightly, with the same results

**Comment by rebron** [comment]
@ryanbr macOS? I think we have a fix coming in here: https://github.com/brave/brave-browser/issues/10724

**Comment by ryanbr** [comment]
Testing in Windows here @rebron

**Comment by simonhong** [comment]
I can get precise location with Brave beta on Win10 after turn wifi on and also turned `Allow apps to access your location` on from system settings. W/o wifi, it's a little bit inaccurate and similar on chrome also.

**Comment by bsclifton** [comment]
For this, we should be able to check if `Location services` and also `Let apps access your location` are enabled
![image](https://github.com/brave/brave-browser/assets/4733304/fc9968fd-554e-43f6-be3d-9c54e33a1723)

If these are not enabled, we can show a modal making clear that the results may not be accurate (with a `Don't show this again` checkbox). We can include a link for how to enable / read more about this:
https://support.microsoft.com/en-us/windows/windows-location-service-and-privacy-3a8eee0a-5b0b-dc07-eede-2a5ca1c49088

When enabled, Brave will show in the list like this:
![image](https://github.com/brave/brave-browser/assets/4733304/e81a457c-3624-49ec-ae70-79272b8188ea)

**Comment by simonhong** [comment]
On windows, we could get location service enabling state with below method.
```
#include <windows.devices.enumeration.h>
#include <wrl/event.h>

#include "base/win/core_winrt_util.h"

namespace {

using ABI::Windows::Devices::Enumeration::DeviceAccessStatus;
using ABI::Windows::Devices::Enumeration::DeviceClass;
using ABI::Windows::Devices::Enumeration::IDeviceAccessInformation;
using ABI::Windows::Devices::Enumeration::IDeviceAccessInformationStatics;
using Microsoft::WRL::ComPtr;

bool IsSystemLocationSettingEnabled() {
  ComPtr<IDeviceAccessInformationStatics> dev_access_info_statics;
  HRESULT hr = base::win::GetActivationFactory<
      IDeviceAccessInformationStatics,
      RuntimeClass_Windows_Devices_Enumeration_DeviceAccessInformation>(
      &dev_access_info_statics);
  if (FAILED(hr)) {
    VLOG(1) << "IDeviceAccessInformationStatics failed: " << hr;
    return true;
  }

  ComPtr<IDeviceAccessInformation> dev_access_info;
  hr = dev_access_info_statics->CreateFromDeviceClass(
      DeviceClass::DeviceClass_Location, &dev_access_info);
  if (FAILED(hr)) {
    VLOG(1) << "IDeviceAccessInformation failed: " << hr;
    return true;
  }

  auto status = DeviceAccessStatus::DeviceAccessStatus_Unspecified;
  dev_access_info->get_CurrentStatus(&status);

  return !(status == DeviceAccessStatus::DeviceAccessStatus_DeniedBySystem ||
           status == DeviceAccessStatus::DeviceAccessStatus_DeniedByUser);
}
```

**Comment by simonhong** [comment]
When we use another dialog for notifying users about location service disabled state,
then browser launches two dialogs simultaneously - permission request and this new service enabling ask dialog.
Is this ok from UX point of view? @rebron @bsclifton 
Or how about adding additional text to permission request bubble?

**Comment by simonhong** [comment]
`GeoLocationImproveTabHelper` monitors each tab's  permission request state and showing UI when currently active tab got geolocation permission and system location service is not enabled.
When a tab is activated, tab helper could show UI if that tab gets geolocation permission request but UI is not launched for current loading.

**Comment by simonhong** [comment]
When geolocation permission is needed, our info dialog is launched and then permission bubble is shown after our dialog is closed.

Our dialog's contents will be like below.
```
Geolocation with Brave

The Brave browser is attempting to access your location. You may not get accurate results unless you enable your operating system's `Location services` setting and also enable `Let apps access your location`.

With those enabled, your location results should be much more accurate

Learn more (link to https://support.microsoft.com/en-us/windows/windows-location-service-and-privacy-3a8eee0a-5b0b-dc07-eede-2a5ca1c49088)

[ ] Don't show this again

OK
```

@Brave-Matt can you check above learn more link?

**Comment by aguscruiz** [comment]
@simonhong I have a design here for how dialogs in general should look like, this should be the default.

https://www.figma.com/file/tU9zrk9Ite8B06JSjHAHhn/General-dialogs-and-popups?type=design&node-id=113%3A17205&mode=design&t=GzxS31F8wW3RJcMl-1

And here's the library file if you need
https://www.figma.com/file/H11ZOl6JMYbCTW4ZJXqR5V/%F0%9F%A6%81-Browser?type=design&node-id=2178%3A22564&mode=design&t=jfEQrWCzF0bhJxFR-1

**Comment by goodov** [comment]
I don't like the path this is going currently. The permission decision step is pretty important, and having additional dialogs doesn't look right at this point. Can we use the omnibox area to nag the user gently AFTER the permission was granted? It should be fine to have the first location be not perfect and give the user ability to improve it afterwards if they wants to. 

There's another concern that enabling this Windows feature by default will allow most (all?) non-Store apps use the precise location without user noticing. Can we be more clear here about the possible downsides? This looks weird pushing this too explicitly when most people might not need it.

cc @pes10k @fmarier to maybe have their opinion on this.

https://github.com/brave/brave-browser/assets/5928869/1cd0144c-601f-4ccb-89a5-26d8a41b3033

**Comment by pes10k** [comment]
Thanks for the heads up @goodov 

Quick thoughts and suggestions

1. I suggest rewording this from "You may not get accurate results unless you enable your operating system's `Location services` setting and also enable `Let apps access your location`." to something more like "By default, Brave protects your privacy by only allowing sites with the geo-location permission to learn about your general location.  If you would like to share your precise location with sites, you can do so by X"
2. I agree with @goodov that having two dialogs looks very odd. Can't this be collapsed into a single dialog
3. More importantly, it seems critical to integrate the "high resolution location" capability into the permission system, so that permissions are partitioned, optionally life-time limited, etc

**Comment by alcarraz** [comment]
I have the same problem on Linux, I have location services enabled and all the other applications that need it work ok, except for brave.

**Comment by fmarier** [comment]
@alcarraz We are not yet using the OS location service on Linux, see https://github.com/brave/brave-browser/issues/29715.

**Comment by alcarraz** [comment]
Ok, but how can I make Brave identify my location, it's kilometers away.

**Comment by fmarier** [comment]
@alcarraz That would depend on the website you're using (e.g. typing in an address or postal/zip code). At the moment, Brave on Linux can only provide to the website a location estimate that's based on your IP address.

**Comment by alcarraz** [comment]
The problem is that, that location estimate is always wrong, and that's for
all websites that use location and all those work well in every single
other browser I tried, even chromium in which brave is based.

----
Enviado desde mi móvil, disculpas por la brevedad.

Sent from my cellphone, I apologize for the brevity.

Andrés Alcarraz.

**Comment by alcarraz** [comment]
I was going to tests the Mozilla Location Service and when I tried to get an API key I found this https://github.com/mozilla/ichnaea/issues/2065. I wanted to know if my location is so wrong because it is what MLS says. I arrived three days late :(

**Comment by stephendonner** [comment]
## Verified `IN-PROGRESS` using

```
Brave | 1.67.82 Chromium: 125.0.6422.41 (Official Build) beta (x86_64)
-- | --
Revision | cbebefd4c702801af65a196ba9701a907ed0d292
OS | macOS Version 11.7.10 (Build 20G1427)
```

### Steps:
Launch browser and open inspector with any tabs
Run navigator.geolocation.getCurrentPosition(() => {}, () => {}, { enableHighAccuracy : false });
check Type A bubble is shown
Dismiss bubble and reset permission state if needed
Disable OS location service
Run navigator.geolocation.getCurrentPosition(() => {}, () => {}, { enableHighAccuracy : true });
Chek Type B bubble is shown
Dismiss bubble and reset permission state if needed
Enable OS location service (On macOS, brave should be enabled also)
Run navigator.geolocation.getCurrentPosition(() => {}, () => {}, { enableHighAccuracy : true });
Check Type C bubble is shown

**Comment by MadhaviSeelam** [comment]
Verification `PASSED` using
```
Brave | 1.67.86 Chromium: 125.0.6422.60 (Official Build) beta (64-bit)
-- | --
Revision | a1f8e8f462d422d4b924fa9c34246f68862b21ad
OS | Windows 11 Version 23H2 (Build 22631.3593)
```

1. installed 1.67.86
2. launch browser
3. opened Developer Tools' Console tab
4. ran navigator.geolocation.getCurrentPosition(() => {}, () => {}, { enableHighAccuracy : false }); (after allow pasting)
5. confirmed Type A permission-prompt dialog was shown with general location request
6. dismissed the dialog (reset permission state if needed)
7. disabled OS location service
8. ran navigator.geolocation.getCurrentPosition(() => {}, () => {}, { enableHighAccuracy : true });
9. confirmed Type C bubble is shown
10. dismissed dialog and reset permission state if needed
11. enabled OS location service, and ensured Brave Beta was checked
12. ran navigator.geolocation.getCurrentPosition(() => {}, () => {}, { enableHighAccuracy : true });
13. confirmed Type B bubble is shown

step 5 (Type A) | step 7 | step 9 (Type C) | step 11 | step 13 (Type B)
--- | ---- | ----- | ----- | ------  
![image](https://github.com/brave/brave-browser/assets/98358127/8180cfaf-da7e-4b79-87fa-3aeda0421b22)|![image](https://github.com/brave/brave-browser/assets/98358127/7cbcaa63-3f53-46c4-924d-7a9798a619fa)|![image](https://github.com/brave/brave-browser/assets/98358127/e6d15219-e338-4d6b-a1a9-8fe1785065a9)|![image](https://github.com/brave/brave-browser/assets/98358127/a3578f59-ab7d-42ac-afd4-49afa784bcbd)|![image](https://github.com/brave/brave-browser/assets/98358127/93b3bad2-78a1-4790-9856-39c4d77fda41)
